### PR TITLE
Optimizely Feature Experimentation docs

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Added cloud mode documentation for Optimizely Feature Experimentation (Full Stack)
> Updated and restructured the existing docs for device mode.

## Type of documentation

- [x] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+| `event` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `eventKey` | String |
+| `userId` / `anonymousId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `userId` | String |
+| `traits` / `context.traits` | `userAttributes` | Object |
+| `properties` | `eventTags` | Object |
+
+### Viewing experimentation results
+
+To view the experimentation-related metrics in the **Results** page of your Optimizely dashboard, make sure you:
+
+- Use the <a href="https://docs.developers.optimizely.com/experimentation/v3.1.0-full-stack/reference/activate">`activate`</a> API first and then send your `track` events. If you do not activate the user, RudderStack will not send any decision event to Optimizely and the event won't be attributed to the user.
```
Comment (Gauravudia) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:44]: Here, RudderStack will still send the track event (decision) but If activate call is not made prior to the track call then the event will not be visible on the result page

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+- Pass the same `userId` in your events that you pass to the `activate` API.
+
+## Page
```
Comment (Gauravudia) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:47]: RudderStack is internally using Track call (line 11) with the event name **Viewed ${category} page** or **Viewed ${name} page**. So, all property defined in Track call (line 11) is applicable for Page call as well.

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+A sample `page` call is shown below:
+
+```javascript
+rudderanalytics.page("home");
+```
+
+In the above snippet, `home` is the name of the web page. If **Track Named Pages** setting is enabled, RudderStack sends a `track` event named `Viewed home page` to Optimizely.
+
+## Screen
```
Comment (Gauravudia) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:61]: RudderStack is internally using Track call (line 11) with the event name **Viewed ${name} screen** .So, all property defined in Track call (line 11) is applicable for Screen call as well.

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx (hunk 2 above)
Comment (Gauravudia) [reply docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:47]: We can provide a reference of Track call (line 11) here.

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx (hunk 3 above)
Comment (Gauravudia) [reply docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:61]: We can provide a reference of Track call (line 11) here.

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx (hunk 1 above)
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:44]:
```suggestion
- Use the <a href="https://docs.developers.optimizely.com/experimentation/v3.1.0-full-stack/reference/activate">`activate`</a> API first and then send your `track` events. If you do not activate the user, RudderStack will send the `track` events to Optimizely and they will not be visible on the **Results** page.
```

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+### Viewing experimentation results
+
+To view the experimentation-related metrics in the **Results** page of your Optimizely dashboard, make sure you:
+
+- Use the <a href="https://docs.developers.optimizely.com/experimentation/v3.1.0-full-stack/reference/activate">`activate`</a> API first and then send your `track` events. If you do not activate the user, RudderStack will not send any decision event to Optimizely and the event won't be attributed to the user.
+- Pass the same `userId` in your events that you pass to the `activate` API.
+
+## Page
+
+RudderStack's <Link to="/event-spec/standard-events/page">`page`</Link> call lets you record your website's page views with any additional relevant information about the viewed page.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:49]:
```suggestion
You can use the <Link to="/event-spec/standard-events/page">`page`</Link> call to record your website's page views with any additional relevant information about the viewed page.
```

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+If you have enabled the **Track Categorized Pages** or **Track Named Pages** dashboard setting, then RudderStack sends a `page` call with the page `category` or `name` as a `track` event to Optimizely with the event name as `Viewed ${category} page` or `Viewed ${name} page`.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:51]:
```suggestion
If you have enabled the **Track Categorized Pages** or **Track Named Pages** dashboard setting, then RudderStack sends a `track` call with the page `category` or `name` to Optimizely, with the event name as `Viewed ${category} page` or `Viewed ${name} page`.
```

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:52]:
```suggestion

<div class="infoBlock">
All the mappings supported for the <Link to="#supported-mappings">`track`</Link> event are applicable for `page` event as well.
</div>

```

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+```javascript
+rudderanalytics.page("home");
+```
+
+In the above snippet, `home` is the name of the web page. If **Track Named Pages** setting is enabled, RudderStack sends a `track` event named `Viewed home page` to Optimizely.
+
+## Screen
+
+RudderStack's <Link to="/event-spec/standard-events/screen">`screen`</Link> call lets you record your users' mobile screen views with any additional relevant information about the screen.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:63]:
```suggestion
You can use the <Link to="/event-spec/standard-events/screen">`screen`</Link> call to record your users' mobile screen views with any additional relevant information about the screen.
```

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+RudderStack sends every `screen` call to Optimizely as a `track` event with the name `Viewed ${name} screen`, where `name` is the name of the viewed screen. If you do not provide any name, RudderStack sets the event name as `Viewed screen`.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:65]:
```suggestion
RudderStack sends every `screen` call you make as a <Link to="#track">`track`</Link> event to Optimizely,   with the name `Viewed ${name} screen` - where `name` is the name of the viewed screen. If you do not provide any name, RudderStack sets the event name as `Viewed screen`.
```

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:66]:
```suggestion

<div class="infoBlock">
All the mappings supported for the <Link to="#supported-mappings">`track`</Link> event are applicable for `page` event as well.
</div>

```

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:44]: Got it - made the change accordingly.

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx (hunk 2 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:47]: Understood - made the necessary changes to reflect this.

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx (hunk 3 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:61]: Understood - made the necessary changes to reflect this.

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+- Pass the same `userId` in your events that you pass to the `activate` API.
+
+## Page
+
+You can use the <Link to="/event-spec/standard-events/page">`page`</Link> call to record your website's page views with any additional relevant information about the viewed page.
+
+If you have enabled the **Track Categorized Pages** or **Track Named Pages** dashboard setting, then RudderStack sends every `page` call along with the page `category` or `name` as a <Link to="#track">`track`</Link> event to Optimizely, with the event name as `Viewed ${category} page` or `Viewed ${name} page`.
+
+<div class="infoBlock">
+All the mappings supported for the <Link to="#supported-mappings">`track`</Link> event are applicable for `page` event as well.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:54]:
```suggestion
All the <Link to="#supported-mappings">mappings</Link> supported for the <code class="inline-code">track</code> event are applicable for <code class="inline-code">page</code> event as well.
```

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+In the above snippet, `home` is the name of the web page. If **Track Named Pages** setting is enabled, RudderStack sends a `track` event named `Viewed home page` to Optimizely.
+
+## Screen
+
+You can use the <Link to="/event-spec/standard-events/screen">`screen`</Link> call to record your users' mobile screen views with any additional relevant information about the screen.
+
+RudderStack sends every `screen` call you make as a <Link to="#track">`track`</Link> event to Optimizely, with the name `Viewed ${name} screen` - where `name` is the name of the viewed screen. If you do not provide any name, RudderStack sets the event name as `Viewed screen`.
+
+<div class="infoBlock">
+All the mappings supported for the <Link to="#supported-mappings">`track`</Link> event are applicable for `screen` event as well.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:72]:
```suggestion
All the <Link to="#supported-mappings">mappings</Link> supported for the <code class="inline-code">track</code> event are applicable for <code class="inline-code">screen</code> event as well.
```

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx (hunk 1 above)
Comment (Gauravudia) [reply docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx]: Let's rephrase it
`Use the <a href="https://docs.developers.optimizely.com/experimentation/v3.1.0-full-stack/reference/activate">`activate`</a> API first and then send your `track` events. If you do not activate the user, the event won't be attributed to the user, therefore the event won't be shown in the Results Page`

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+| `event` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `eventKey` | String |
+| `userId` / `anonymousId` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span> | `userId` | String |
+| `traits` / `context.traits` | `userAttributes` | Object |
+| `properties` | `eventTags` | Object |
+
+### Viewing experimentation results
+
+To view the experimentation-related metrics in the **Results** page of your Optimizely dashboard, make sure you:
+
+- Use the <a href="https://docs.developers.optimizely.com/experimentation/v3.1.0-full-stack/reference/activate">`activate`</a> API first and then send your `track` events. If you do not activate the user, RudderStack will send the `track` events to Optimizely and they will not be attributed to the user (visible on the **Results** page).
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx:44]:
```suggestion
- Use the <a href="https://docs.developers.optimizely.com/experimentation/v3.1.0-full-stack/reference/activate">`activate`</a> API first and then send your `track` events. If you do not activate the user, the event won't be attributed to the user and won't be shown in the **Results** page.
```

File: docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/optimizely-feature-experimentation/cloud-mode.mdx]: Done - slightly modified the last sentence for better flow and clarity.

**Review by Gauravudia — APPROVED** [review]

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**